pluginconfig/execute.go

```diff
@@ -78,6 +81,128 @@ func (e ExecuteOffchainConfig) Validate() error {
+	// Example would be CCTPv1 using AttestationAPI and CCTPv2 using a different API or completely
+	// different strategy which requires different configuration and implementation during Observation phase.
+	// [
+	//  {
+	//    "type": "usdc-cctp",
+	//    "version": "1.0",
+	//    "attestationAPI": "http://circle.com/attestation",
+	//    "attestationAPITimeout": "1s",
+	//    "attestationAPIIntervalMilliseconds": "500ms"
+	//  },
+	//  {
+	//    "type": "usdc-cctp",
+	//    "version": "2.0",
+	//    "customCirlceAPI": "http://cirle.com/gohere",
+	//    "yetAnotherAPI": "http://cirle.com/anotherone",
+	//    "customCircleAPITimeout": "1s",
+	//    "yetAnotherAPITimeout": "500ms"
+	//  }
+	//]
+	// Having version in that JSONisn't expensive, but it could reduce the risk of breaking the observers in the future.
+	Version string `json:"version"`
+
+	*USDCCCTPObserverConfig
+}
+
+// WellFormed checks that the observer's config is syntactically correct - proper struct is initialized based on type
+func (t TokenDataObserverConfig) WellFormed() error {
+	switch t.Type {
+	case USDCCCTPHandlerType:
+		if t.USDCCCTPObserverConfig == nil {
+			return errors.New("USDCCCTPObserverConfig is empty")
+		}
+	default:
+		return errors.New("unknown token data observer type")
+	}
+	return nil
+}
+
+// Validate checks that the observer's config is semantically correct - fields are set correctly
+// depending on the config's type
+func (t TokenDataObserverConfig) Validate() error {
+	switch t.Type {
+	case USDCCCTPHandlerType:
+		if err := t.USDCCCTPObserverConfig.Validate(); err != nil {
+			return err
+		}
+	default:
+		return errors.New("unknown token data observer type " + t.Type)
+	}
+	return nil
+}
+
+const USDCCCTPHandlerType = "usdc-cctp"
+
+type USDCCCTPObserverConfig struct {
+	Tokens                 map[int]USDCCCTPTokenConfig `json:"tokens"`
+	AttestationAPI         string                      `json:"attestationAPI"`
+	AttestationAPITimeout  *commonconfig.Duration      `json:"attestationAPITimeout"`
+	AttestationAPIInterval *commonconfig.Duration      `json:"attestationAPIInterval"`
+}
+
+func (p USDCCCTPObserverConfig) Validate() error {
+	if p.AttestationAPI == "" {
+		return errors.New("AttestationAPI not set")
+	}
+	if len(p.Tokens) == 0 {
+		return errors.New("Tokens not set")
+	}
+	for _, token := range p.Tokens {
+		if err := token.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// nolint:lll // CCTP link
+type USDCCCTPTokenConfig struct {
+	// SourceTokenAddress is the address of the USDC ECR-20 token on the source chain
+	SourceTokenAddress string `json:"sourceTokenAddress"`
+	// SourceMessageTransmitterAddr is the address of the CCTP MessageTransmitter address on the source chain
+	// https://github.com/circlefin/evm-cctp-contracts/blob/adb2a382b09ea574f4d18d8af5b6706e8ed9b8f2/src/MessageTransmitter.sol
+	SourceMessageTransmitterAddr string `json:"sourceMessageTransmitterAddress"`
```

Comment:
General comment not specific to this PR - we really need a proper address type to abstract the different ways we need to encode addresses.